src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcEndpointInfo.cs

```diff
@@ -51,14 +51,14 @@ internal class MvcEndpointInfo
 
         public IDictionary<string, IRouteConstraint> Constraints { get; }
         public RouteValueDictionary DataTokens { get; }
-        internal RouteTemplate ParsedTemplate { get; private set; }
+        internal RoutePattern ParsedPattern { get; private set; }
 
         private static IDictionary<string, IRouteConstraint> GetConstraints(
             IInlineConstraintResolver inlineConstraintResolver,
-            RouteTemplate parsedTemplate,
+            RoutePattern parsedTemplate,
             IDictionary<string, object> constraints)
         {
-            var constraintBuilder = new RouteConstraintBuilder(inlineConstraintResolver, parsedTemplate.TemplateText);
+            var constraintBuilder = new RouteConstraintBuilder(inlineConstraintResolver, parsedTemplate.RawText);
```

Comment:
nit: Could you add a named argument for raw text here as its not clear that its being used as a display name